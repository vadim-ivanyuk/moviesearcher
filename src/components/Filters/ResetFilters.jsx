import React from "react";
import { useDispatch } from "react-redux";
import { resetFilters } from "../../redux/filters/filters.actions";

export const ResetFilters = () => {
  const dispatch = useDispatch();

  const handelClick = () => {
    dispatch(resetFilters());
  };

  return (
    <button className="btn btn-light col-12 mt-3 reset-filters" onClick={handelClick}>
      Сбросить фильтры
    </button>
  );
};
