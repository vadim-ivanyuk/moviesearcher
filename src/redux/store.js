import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";
import rootReducer from "./rootReducer";
import { ON_CHANGE_FILTERS } from "./filters/filters.types";

let currentPage;

const onChangePageMiddleware = ({ dispatch, getState }) => (next) => (
  action
) => {
  currentPage = getState().filters.page;
  if (
    action.type === ON_CHANGE_FILTERS &&
    action.payload.name !== "page" &&
    currentPage !== 1
  ) {
    dispatch({
      type: ON_CHANGE_FILTERS,
      payload: {
        name: "page",
        value: 1,
      },
    });
  }

  return next(action);
};

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk, onChangePageMiddleware))
);

export default store;
