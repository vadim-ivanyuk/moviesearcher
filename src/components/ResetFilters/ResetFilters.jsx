import React from "react";
import { useDispatch } from "react-redux";
import { resetFilters } from "../../redux";

export const ResetFilters = () => {
  const dispatch = useDispatch();

  const handelResetFilters = () => {
    dispatch(resetFilters());
  };

  return (
    <button
      className="btn btn-light col-12 mt-3 reset-filters"
      onClick={handelResetFilters}
    >
      Сбросить фильтры
    </button>
  );
};
