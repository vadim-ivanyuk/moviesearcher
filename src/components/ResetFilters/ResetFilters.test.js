import React from "react";
import { configure, mount } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import { Provider } from "react-redux";
import store from "../../redux/store";
import { ResetFilters } from "./ResetFilters";

configure({ adapter: new Adapter() });

describe("Reset filters", () => {
  const resetFilters = mount(
    <Provider store={store}>
      <ResetFilters />
    </Provider>
  );

  it("snap Reset filters", () => {
    expect(shallowToJson(resetFilters)).toMatchSnapshot();
  });
});
