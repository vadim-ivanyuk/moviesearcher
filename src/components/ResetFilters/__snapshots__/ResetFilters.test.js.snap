// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset filters snap Reset filters 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ResetFilters>
    <button
      className="btn btn-light col-12 mt-3 reset-filters"
      onClick={[Function]}
    >
      Сбросить фильтры
    </button>
  </ResetFilters>
</Provider>
`;
